OrthoEvol/Manager/database_management.py

```diff
@@ -1,682 +1,808 @@
+                    stdout = self.db_mana_utils.system_cmd(cmd=cmd, stdout=sp.PIPE, stderr=sp.STDOUT, shell=True)
+        else:
+            self.db_mana_log.critical("Please set the BLAST environment variables in your .bash_profile!!")
+            self.db_mana_log.info("The appropriate environment variable is \'BLASTDB=%s\'." % str(dl_path))
+            self.db_mana_log.critical("Please set the BLAST environment variables in your .bash_profile!!")
+
+    def download_ete3_taxonomy_database(self):
+        """Update ETE3's taxonomy database with ETE3's API."""
+        # DEFAULT_TAXADB = os.path.join(os.environ.get('HOME', '/'), '.etetoolkit', 'taxa.sqlite')
+        ete3 = import_module("ete3")
+        ncbi_taxon_dump_path = self.database_path / Path("NCBI") / Path('pub') / Path('taxonomy')
+        ncbi = ete3.NCBITaxa(dbfile=str(ncbi_taxon_dump_path / 'ete3_taxa.sqlite'))
+        ncbi.update_taxonomy_database()
+
+    def copy_biosql_taxonomy_template(self, destination, database_name):
+        """
+        Copy a BioSQL template database loaded with NCBI's taxonomy data.
+        :param destination:  Where the template will be copied to.
+        :type destination:  str.
+        :param database_name:  The name of the copied database.
+        :type database_name:  str.
+        """
+        if self.driver.lower() == "sqlite3":
+            ncbi_db = self.biosql.SQLiteBioSQL(database_name=database_name,
+                                               proj_mana=self.proj_mana)
+            ncbi_db.copy_template_database(destination=destination)
+            return ncbi_db
+
+        elif self.driver.lower() == "mysql":
+            # db_path = self.ncbi_db_repo / Path('pub') / Path('taxonomy')
+            # ncbi_db = self.biosql.MySQLBioSQL()
+            # return ncbi_db
+            pass
+
+    def create_biosql_taxonomy_template(self):
+        """Creates a template database by uploading SQLite schema and NCBI taxonomy."""
+        if self.driver.lower() == "sqlite3":
+            ncbi_db = self.biosql.SQLiteBioSQL(proj_mana=self.proj_mana)
+            ncbi_db.create_template_database()
+        return ncbi_db
+
+    def download_itis_taxonomy_tables(self):
+        # Loads data from ITIS via http://www.itis.gov/downloads/
+        # Use this along with BioSQL's phyloDB
+        pass
+
+    def download_ncbi_taxonomy_dump_files(self, url='''ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz'''):
+        """Download and extract the NCBI taxonomy dump files via a GET request.
+
+        :param url: A ftp link to the NCBI taxdump*.tar.gz file of interest.
+        :type url: str.
+        """
+        # TODO: Add this to NCBI FTP client
+        dl_path = Path(self.database_path) / Path("NCBI") / Path('pub') / Path('taxonomy')
+        dl_abs_path = dl_path / Path('taxdump.tar.gz')
+        r = urllib.request.urlopen(url)
+        data = r.read()
+        print('taxdump.tar.gz')
```

Comment:
Log this or remove line. I'm not sure if this is being logged elsewhere.

Reply:
That was a debugging mistake.

```suggestion
```